### DISCUSSION TITLE
fetch_record_batch dropping timezone information from timestamps (fixed in 0.8.1)

### DISCUSSION THREAD

**Discussion by nph** [duckdb/duckdb#9100] [General]
Thought I'd post this here just in case it's helpful. Today I discovered a bug where `fetch_record_batch` drops the timezone from timezone-aware timestamps. It appears to impact all versions of DuckDB from 0.6.1 to 0.8 (didn't test prior to 0.6.1). It's fixed in 0.8.1, which is great, but I was curious about the fix since nothing in the release [notes](https://github.com/duckdb/duckdb/releases/tag/v0.8.1) mentions timezones or `fetch_record_batch`.

## To Reproduce
```
import duckdb
import pandas as pd
import pyarrow as pa

con = duckdb.connect(database=':memory:')

# Create a DF containing timezone-aware timestamps
df = pd.DataFrame(pd.date_range(start='1/1/2023', periods=10, tz='Asia/Tokyo'),
                  columns=['datetime_with_tz'])

# Confirm the timezone is present
print(df.dtypes)
# datetime_with_tz    datetime64[ns, Asia/Tokyo]

# Query and return results as a PyArrow Table directly
table = con.execute('SELECT * from df').fetch_arrow_table()

# Check the schema - all good
print(table.schema)
# datetime_with_tz: timestamp[us, tz=UTC]

# Query and return results as a PyArrow Table via a record batch reader
table2 = pa.Table.from_batches(con.execute('SELECT * from df').fetch_record_batch())

# Now the schema is missing the time-zone...
print(table2.schema)
# datetime_with_tz: timestamp[us]
```

## OS:
MacOS

## DuckDB Version:
0.6.1 - 0.8.0

## DuckDB Client:
Python

## Full Name:
Nick Hughes

## Affiliation:
Interval IQ

**Comment by Tishj** [comment]
It's likely a side-effect of the fix that addressed arrow LargeString's

**Comment by nph** [comment]
Thanks @Tishj
